### PR TITLE
doc improvement: Replace SPM with TFM in the dependencies for nRF9160 samples

### PR DESCRIPTION
**Pull request by divipillai**
Updated the dependencies section to replace SPM with TFM for the
nRF9160 samples which supports TFM instead of SPM.
Following are the samples which are effected:
at_client
at_monitor
aws_fota
azure_fota
azure_iot_hub
coap_client
download
fmfu_smp_svr
https_client
http_update\application_update
http_update\full_modem_update
http_update\modem_delta_update
location
lte_ble_gateway
modem_callbacks
mqtt_simple
pdn

Updated the 'Building and running' section to remove the mention about TFM as it supports SPM: 
gnss
lwm2m_client
modem_shell
secure_services

updated ciphersuites sample with TFM support.

Signed-off-by: divya pillai <divya.pillai@nordicsemi.no>

### CODE REVIEW

**Review by coderbyheart — APPROVED** [review]

**Comment by divipillai** [comment]
@b-gent and @shantha-14 - We could consider this for 2.0 release.

**Review by b-gent — APPROVED** [review]

File: samples/nrf9160/secure_services/README.rst [hunk 1]
```diff
@@ -37,6 +37,6 @@ Building and running
 Dependencies
 ************
 
-This sample uses the following |NCS| libraries:
+This sample uses the following secure firmware component:
 
-* :ref:`lib_spm`
+* :ref:`Trusted Firmware-M <ug_tfm>`
```
Comment (divipillai) [inline samples/nrf9160/secure_services/README.rst:42]: @oyvindronningstad : Is this true? Does the secure_services sample support TFM, by default. I could see the mention about this in line 20. But the sample describes about SPM. Could you please confirm this.

Comment (oyvindronningstad) [reply samples/nrf9160/secure_services/README.rst:42]: No, the secure_services sample is SPM-only, because of API differences.

**Review by joerchan — APPROVED** [review]

File: samples/nrf9160/secure_services/README.rst (hunk 1 above)
Comment (joerchan) [reply samples/nrf9160/secure_services/README.rst:42]: This is an SPM sample.

**Review by joerchan — CHANGES_REQUESTED** [review]

**Comment by divipillai** [comment]
> It should probably be documented somewhere that SPM has been enabled for all Thingy91 builds, ref:
> 
> #7781

We have documented it in the applications and samples that supports Thingy:91 - https://github.com/nrfconnect/sdk-nrf/pull/7822.

**Review by joerchan — APPROVED** [review]

**Comment by b-gent** [comment]
> > It should probably be documented somewhere that SPM has been enabled for all Thingy91 builds, ref:
> > #7781
> 
> We have documented it in the applications and samples that supports Thingy:91 - #7822.

Hard to find a good place for it, added in 'nRF9160 samples -> Updated' section.
@simensrostad  please confirm this is fine

**Review by jhirsi — APPROVED** [review]

**Review by tokangas — APPROVED** [review]

File: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst [hunk 2]
```diff
@@ -340,7 +340,7 @@ After programming the application and all the prerequisites to your development
 1. |connect_kit|
 #. Connect to the kit with a terminal emulator (for example, LTE Link Monitor). See :ref:`lte_connect` for more information.
 #. Reset the development kit.
-#. Observe in the terminal window that the development kit starts up in the Trusted Firmware-M secure firmware and that the application starts.
+#. Observe in the terminal window that the development kit starts up in the Trusted Firmware-M secure firmware when building with nRF9160 DK or in the Secure Partition Manager when building with thingy:91 and that the application starts.
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst:343]:
```suggestion
#. Observe in the terminal window that the development kit starts up in the Trusted Firmware-M secure firmware when building with nRF9160 DK or in the Secure Partition Manager when building with Thingy:91 and that the application starts.
```

File: doc/nrf/ug_nrf91_features.rst [hunk 3]
```diff
@@ -54,7 +54,7 @@ Application MCU
 
 The M33 TrustZone divides the application MCU into secure and non-secure domains.
 When the MCU boots, it always starts executing from the secure area.
-The secure bootloader chain starts the :ref:`Trusted Firmware-M (TF-M) <ug_tfm>` (or the :ref:`nrf9160_ug_secure_partition_manager`) which configures a part of memory and peripherals to be non-secure, and then jumps to the main application located in the non-secure area.
+The secure bootloader chain starts the :ref:`Trusted Firmware-M (TF-M) <ug_tfm>` when using nRF9160 DK or the :ref:`nrf9160_ug_secure_partition_manager` when using thingy:91 which configures a part of memory and peripherals to be non-secure, and then jumps to the main application located in the non-secure area.
```
Comment (umapraseeda) [inline doc/nrf/ug_nrf91_features.rst:57]:
```suggestion
The secure bootloader chain starts the :ref:`Trusted Firmware-M (TF-M) <ug_tfm>` when using nRF9160 DK or the :ref:`nrf9160_ug_secure_partition_manager` when using Thingy:91.
The secure firmware component configures a part of memory and peripherals to be non-secure, and then jumps to the main application located in the non-secure area.
```

**Review by jtguggedal — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]

**Comment by divipillai** [comment]
> > It should probably be documented somewhere that SPM has been enabled for all Thingy91 builds, ref:
> > #7781
> 
> We have documented it in the applications and samples that supports Thingy:91 - #7822.

@simensrostad - I have now updated the nRF91 user guide as well with these changes.

**Comment by divipillai** [comment]
@MirkoCovizzi : Could you review the changes for the ciphersuites sample.

**Review by MirkoCovizzi — APPROVED** [review]
Approving with regards to ciphersuites sample, cannot attest for the rest.

**Pull request merged**